### ISSUE TITLE
Make an enum in e2e tests to wrap around custom errors

### ISSUE THREAD

**Issue by vidyaap** [boltlabs-inc/key-mgmt#86]
The e2e tests currently check for failures in each running process (client library, key server) by explicitly specifying an expected error in the `Outcome` of the test. On running an operation, the test execution compares the actual error (if any) to the expected error.

However, right now this comparison is a bit awkward. The `expected_error` is an `Option<anyhow::Error>`, and we currently manually hard-code the expected error text in each test specification, e.g.
```rust
expected_error: Some(anyhow!("AuthenticationFailed")),
```
This is not ideal, since any change to the error types (e.g. enum variant names) or error text will cause the test to fail.

One improvement would use a public error enum/type from our library as the type of the `expected_error`. This would allow us to use the enum variant directly in the test specification. However, this is not quite correct because different running processes (local client library, remote client library, key server) might use _different_ public error types. 
```rust
expected_error: Some(LocalClientError::AuthenticationFailed),
```

It would be better to have an enum in the tests that wraps around our custom error types so we can compare them directly. This might simplify the `Outcome` struct, since we could specify both the error and the user in a single field (rather than two separate ones like now). 
```rust
expected_error: Some(ErrorParty::LocalClient(LocalClientError::AuthenticationFailed)),
```

A related issue: in the match statement that checks "[whether the process errors matched the expectation,](https://github.com/boltlabs-inc/key-mgmt/blob/develop/dams-tests/tests/main.rs#L269-L285)", the `assert_eq` between the expected error and actual error is a bit confusing to me. We should make sure this is the simplest way to compare errors. We should also replace the `assert_eq` with a returned error, so that all tests will run (the current behavior will abort the run at the first failing test).

This issue can be completed when we:
- [ ] Update the expected error to use an error variant rather than a hard-coded anyhow error (and built any extra infra to support all possible variants)
- [ ] Double-check the error-comparison match statement in `execute` to make sure it's as simple as possible 
- [ ] Replace the `assert_eq` that compares expected errors with a returned error

**Comment by indomitableSwan** [comment]
Tagging this as an issue that needs to be fleshed out. @marsella

**Comment by marsella** [comment]
> Tagging this as an issue that needs to be fleshed out. @marsella

I expanded the issue description.

**Comment by picklenerd** [comment]
I think we've got a slightly better setup now where we specify errors by `LockKeeperClientError` but still do a string comparison so we can continue to use `anyhow`. I'm going to close this ticket since the description is out of date and I think this has been resolved in a different way. I added the testing epic to this issue so it will still be easy to find when we start improving the testing framework.
